### PR TITLE
test(cli-e2e): separate e2e tests for lighthouse plugin

### PR DESCRIPTION
**Pull request by BioPhoton**
This PR separates the e2e tests for the lighthous plugin into its own project.

**It includes:**
- ➕ add new project `plugin-lighthouse-e2e`  & 🤌 use `global-setup.verdaccio.ts`
- 📦 use new e2e test setup
- 🧹 move `omitReportData` into `test-utils`. 
- 🧹 move `createNpmWorkspace` into `test-utils`

### CODE REVIEW

File: e2e/create-cli-e2e/tests/init.e2e.test.ts [hunk 1]
```diff
@@ -1,9 +1,9 @@
+import { dirname } from 'knip/dist/util/path';
```
Comment (vmasek) [inline e2e/create-cli-e2e/tests/init.e2e.test.ts:1]: I am not sure if we should be using knip util there
Did you mean to use `import { dirname } from 'node:path';`?

**Review by vmasek — CHANGES_REQUESTED** [review]

File: e2e/create-cli-e2e/tests/init.e2e.test.ts (hunk 1 above)
Comment (BioPhoton) [reply e2e/create-cli-e2e/tests/init.e2e.test.ts]: Uuups! Nice catch! ❤️

**Review by vmasek — DISMISSED** [review]

File: e2e/plugin-lighthouse-e2e/project.json [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    }
+  },
+  "implicitDependencies": [
+    "models",
+    "utils",
+    "core",
+    "cli",
+    "plugin-lighthouse"
+  ],
+  "tags": ["scope:tooling", "type:e2e"]
```
Comment (matejchalk) [inline e2e/plugin-lighthouse-e2e/project.json:28]:
```suggestion
  "tags": ["scope:plugin", "type:e2e"]
```

File: global-setup.verdaccio.ts [hunk 3]
```diff
@@ -32,7 +38,6 @@ export async function setup() {
 export async function teardown() {
   // NOTICE - Time saving optimization
   // We skip uninstalling packages as the folder is deleted anyway
-
   // comment out to see the folder and web interface
-  await nxStopVerdaccioAndTeardownEnv(activeRegistry);
+  // await nxStopVerdaccioAndTeardownEnv(activeRegistry);
```
Comment (matejchalk) [inline global-setup.verdaccio.ts:42]: I guess this shouldn't be commented out by default.

**Review by vmasek — DISMISSED** [review]

**Review by matejchalk — DISMISSED** [review]

**Review by matejchalk — APPROVED** [review]

**Review by vmasek — APPROVED** [review]

**Pull request merged**